### PR TITLE
Stream removes itself upon destruction

### LINKED ISSUE #500: Creating a stream from an EventEmitter should remove listener on destruction
Consider:

```
const ee = new EventEmitter();
const s = highland('someEvent', ee);

typeof ee._events.someEvent; // function

s.destroy();

typeof ee._events.someEvent; // function
```

Given that `highland` adds a new listener to `ee` on creation it should also remove that listener on destruction.

We should be able to use `_destructors` to register the removal:

```
  function eeHandler () {
    var ctx = mapper.apply(this, arguments);
    self.write(ctx);
  }

  ee.on(xs, eeHandler);

  var removeMethod = ee['removeListener'] || ee['unbind'];

  if (removeMethod)
     this._destructors.push(function () {
       ee[removeMethod](xs, eeHandler);
    });
```

### PR DESCRIPTION
**Pull request by briancaine**
Fixes issue #500.

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -2864,6 +2864,16 @@ exports['wrap EventEmitter (or jQuery) on handler'] = function (test) {
     });
 };
 
+exports['removing EventEmitter (or jQuery) listener on destruction'] = function(test) {
```
Comment (vqvu) [inline test/test.js]: Can you make a note of the original issue? Just append " (issue #500)" to the end.

Comment (briancaine) [inline test/test.js]: Here? Issue #500

File: test/test.js [hunk 2]
```diff
@@ -2864,6 +2864,16 @@ exports['wrap EventEmitter (or jQuery) on handler'] = function (test) {
...
+    var ee = new EventEmitter();
+    var s = _('myevent', ee);
+
+    test.same(typeof ee._events.myevent, 'function');
```
Comment (vqvu) [inline test/test.js]: Checking private state like `_events` is brittle. You should listen to the 'removeListener' event instead, and make sure that is is emitted when you destroy the stream.

File: lib/index.js [hunk 3]
```diff
@@ -256,10 +256,20 @@ function __(StreamCtor) {
...
-            secondArg.on(xs, function () {
+
+            var callback_func = function () {
                 var ctx = mapper.apply(this, arguments);
                 s.write(ctx);
-            });
+            };
+
+            secondArg.on(xs, callback_func);
+            var removeMethod = secondArg.removeListener || secondArg.unbind;
```
Comment (vqvu) [inline lib/index.js]: Can you add a comment that says which method belongs to which type of event emitters (EventEmitter or jQuery)?

File: test/test.js (hunk 1 above)
Comment (vqvu) [inline test/test.js]: Sorry, I meant in the test name. LIke [this](https://github.com/caolan/highland/blob/master/test/test.js#L476).

File: test/test.js (hunk 2 above)
Comment (briancaine) [inline test/test.js]: Done.

File: lib/index.js (hunk 3 above)
Comment (briancaine) [inline lib/index.js]: Done.

**Comment by vqvu** [comment]
LGTM.

**Pull request merged**